econplayground/main/views.py

```diff
@@ -591,3 +593,168 @@ def get_success_url(self):
             extra_tags='safe')
 
         return reverse('topic_list', kwargs={'cohort_pk': self.cohort.pk})
+
+
+class AssignmentListView(LoginRequiredMixin, ListView):
+    model = Assignment
+    template_name = 'main/assignment_list.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        if not user_is_instructor(request.user):
+            url = '/accounts/login/'
+            return HttpResponseRedirect(url)
+
+        return super(
+            AssignmentListView, self).dispatch(request, *args, **kwargs)
+
+    def get_queryset(self):
+        return Assignment.objects.filter(
+            instructors=self.request.user).order_by('created_at')
+
+    def get_context_data(self, **kwargs):
+        context = super(AssignmentListView, self).get_context_data(**kwargs)
+
+        # This area should never be reached if this user isn't an instructor,
+        # but just in case.
+        if not user_is_instructor(self.request.user):
+            return context
```

Comment:
This is not a bad approach to some defensive coding, but I would say with LoginRequiredMixin and some unit tests in place, it's really not necessary.

Reply:
Alright. I trust that.